src/parser/mod.rs

```diff
@@ -6013,6 +6013,22 @@ impl<'a> Parser<'a> {
         Ok(expr)
     }
 
+    pub fn parse_operator(&mut self) -> Option<BinaryOperator> {
```

Comment:
Since this is for `ORDER BY ... USING ...` maybe make the name more specific? Something like `parse_using_operator`?

Also feels like a smell that`USING` needs its own operator parsing function, though.

Reply:
I'd have to dig more into how PG handles this itself vs what sqlparser does now for operators to have more concrete suggestions, though.

Reply:
No this function is intended to be a replacement for operator parsing which is a dogs breakfast and buried in the expression parsing stuff. Adding _yet another_ way to parse operators is a bad idea IMHO. Calling this function `parse_using_operator` sets the expectation for future developers that this is just for that purpose and exacerbates the problem.

Reply:
I'm taking the approach of trying to leave the code in a better place than I found it, rather than just throw stuff on the pile.

What's the argument for parsing using operators on their own? Any binary operator is supported AFAICT.

Reply:
> What's the argument for parsing using operators on their own? Any binary operator is supported AFAICT.

Ah, that's not what I was suggesting. I was confused that this is only used for `ORDER BY ... USING ...`. I don't think that this leaves the code in a better place to be honest because it adds a special case to be aware of that's partially implemented.

Might be fine just to unblock us in our fork so that things parse. Probably worth properly designing before raising upstream.